PR: https://github.com/odoo/odoo/pull/173273

From: 88569c97efbb534679700e8c7d56231aa1934b4d
From: Jitendra Prajapat
Date: 2024-08-09 17:05:55

Breaking data model changes scores: del:7 + add:0, change matches:
-    razorpay_authcode = fields.Char('Razorpay APPR Code')
-    razorpay_issuer_card_no = fields.Char('Razorpay Issue Card No Last 4 digits')
-    razorpay_issuer_bank = fields.Char('Razorpay Issuer Bank')
-    razorpay_payment_method = fields.Char('Razorpay Payment Method')
-    razorpay_reference_no = fields.Char('Razorpay Merchant Reference No.')
-    razorpay_card_scheme = fields.Char('Razorpay Card Scheme')
-    razorpay_card_owner_name = fields.Char('Razorpay Card Owner Name')

Total Changes: 7

[IMP] pos*: made pos_payment fields common

pos*: point_of_sale, pos_razorpay, pos_paytm

Before this commit:
===================
There were separate fields for pos_payment for each module, such as pos_paytm,
pos_razorpay, and pos_mercury. Each field stored the same data for each module
individually.

After this commit:
================
A common field for pos_payment are created to store data based on the module,
simplifying the data management and making it easier to display on the UI side.

task- 4011453

Related Upgrade PR: https://github.com/odoo/upgrade/pull/6272

closes odoo/odoo#173273

Related: odoo/upgrade#6272
Signed-off-by: Adrien Guilliams (adgu) <adgu@odoo.com>

================================= pseudo patch: =================================

--- a/addons/pos_razorpay/models/pos_payment.py
+++ b/addons/pos_razorpay/models/pos_payment.py
@@ -4,11 +4,4 @@ class PosPayment(models.Model):
 
     _inherit = "pos.payment"
 
-    razorpay_authcode = fields.Char('Razorpay APPR Code')
-    razorpay_issuer_card_no = fields.Char('Razorpay Issue Card No Last 4 digits')
-    razorpay_issuer_bank = fields.Char('Razorpay Issuer Bank')
-    razorpay_payment_method = fields.Char('Razorpay Payment Method')
-    razorpay_reference_no = fields.Char('Razorpay Merchant Reference No.')
     razorpay_reverse_ref_no = fields.Char('Razorpay Reverse Reference No.')
-    razorpay_card_scheme = fields.Char('Razorpay Card Scheme')
-    razorpay_card_owner_name = fields.Char('Razorpay Card Owner Name')
